docs/content/preview/develop/best-practices-ysql.md

```diff
@@ -119,6 +119,25 @@ Use `BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE READ ONLY DEFERRABLE` for ba
 For more details, see [Large scans and batch jobs](../../develop/learn/transactions/transactions-performance-ysql/#large-scans-and-batch-jobs).
 {{</tip>}}
 
+
```

Comment:
Use the [jsonb](../../api/ysql/datatypes/type_json) datatype to model JSON data; that is, data that doesn't have a set schema but has a truly dynamic schema. jsonb in YSQL is the same as the [jsonb](https://www.postgresql.org/docs/11/datatype-json.html) datatype in PostgreSQL. You can use jsonb to group less interesting or less frequently accessed columns of a table. YSQL also supports jsonb expression indexes, which can be used to speed up data retrieval that would otherwise require scanning the JSON entries.

{{< note title="Use jsonb columns only when necessary" >}}

- A good schema design is to only use jsonb for truly dynamic schema. That is, don't create a "data jsonb" column where you put everything; instead, create a jsonb column for dynamic data, and use regular columns for the other data.
- jsonb columns are slower to read/write compared to normal columns.
- jsonb values take more space because they need to store keys in strings, and maintaining data consistency is harder, requiring more complex queries to get/set jsonb values.
- jsonb is a good fit when writes are done as a whole document with a per-row hierarchical structure. If there are arrays, the choice is not JSONB vs. column, but vs additional relational tables.
- For reads, jsonb is a good fit if you read the whole document and the searched expression is indexed. 
- When reading one attribute frequently, it's better to move it to a column as it can be included in an index for an `Index Only Scan`.

{{< /note >}}

Reply:
Added in https://github.com/yugabyte/yugabyte-db/pull/19233/commits/2770517cb86346fbd782b503500b5c8a893e9137